File: backend/src/main/java/com/grouply/backend/authentication/SignUpRequestDTO.java
package com.grouply.backend.authentication;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import lombok.Data;

@Data
public class SignUpRequestDTO {

    private String firstName;
    private String lastName;
    @Min(6)
    private String username;
    @Email
    private String email;
    @Min(6) @Max(12)
    private String password;
    private String confirmPassword;

}

File: frontend/src/components/pages/authentication/signup-wizard/Dtos/SignUpRequestDTO.ts


type AccountDTO = {
    firstName: string
    lastName: string
    username: string
    email: string
    password: string
    confirmPassword: string
}

type Profile = {

}

export class SignUpRequestDTO {

    accountDetails: AccountDTO

    // Profile related

    constructor(accountDetails: AccountDTO) {
        this.accountDetails = accountDetails
    }
    
}